### ISSUE TITLE
Make plugin compatible with Matomo 4

### ISSUE THREAD

**Issue by Findus23** [RegioneER/RerUserDates#15]
Hi, hope you are doing well!

followup to https://github.com/RegioneER/RerUserDates/issues/12 :)

Matomo 4 is progressing steadily and it would be great if this plugin was also [available for it](https://matomo4-plugins.lw1.at/).

[Migrating the plugin](https://matomo.org/blog/2020/08/matomo4-make-your-plugin-compatible-now/) should be pretty simple as there were not many breaking changes.

You can find the migration guide here: https://developer.matomo.org/guides/migrate-matomo-3-to-4

It boils down to:
- update your dev environment to 4.x-dev
- update your plugin.json (new major version of the plugin and require `>=4.0.0-b1,<5.0.0-b1`)
- check that everything still works fine
- commit this to a new branch (e.g. `4.x-dev`) and make this branch the default in the github settings
- tag and push a new release to publish it to the marketplace

If you later fix a bug and also want to backport it to the 3.x plugin, you can cherry-pick the commit to the old branch, release a new version (lower than the one for 4.x) and publish it and 3.x users should get this update.

If you have questions about anything, just ask me!

**Comment by tassoman** [comment]
Ciao @Findus23 thanks for the reminder, sorry for the late reply, I'm going to adapt the repository in the next hours

**Comment by Findus23** [comment]
Hi,

Putting the whole ajax function into a $(document).ready() solves the weird angular error for me.

```javascript
(function ($, require) {
    $(document).ready(function () {

        var ajaxHelper = require('ajaxHelper');
        var ajax = new ajaxHelper();
[...]
        ajax.send();
    });
})(jQuery, require);
```

You can also use 
```javascript
ajax.addParams({
    module: "API",
    method: "RerUserDates.getSettingsCalendars",
    format: "JSON"
}, "get");
```
instead of 
```javascript
ajax.setUrl("index.php?module=API&method=RerUserDates.getSettingsCalendars&format=JSON");
```
which might be a bit more readable.

**Comment by tassoman** [comment]
Wonderful! I've forgot about the trick.
Thanks for hinting, your help was appreciated!
I'm going to push a refactored version working on 4.x branch

**Comment by tassoman** [comment]
I'm still struggling on the Angular issue I quoted in [Matomo's #15843](https://github.com/matomo-org/matomo/issues/15843). 
The comparison feature was unexpected :roll_eyes:

**Comment by tassoman** [comment]
Fixed in [release 3.0.3](https://github.com/RegioneER/RerUserDates/releases/tag/3.0.3), thank you! :rocket:
